ivadomed/prepare_dataset.py

```diff
@@ -0,0 +1,215 @@
+import nibabel as nib
+import numpy as np
+import matplotlib.pyplot as plt
+import PIL
+import skimage
+import os
+import sys
+from utils import reorient_image
+
+# normalize Image
+def normalize(arr):
+    ma = arr.max()
+    mi = arr.min()
+    return ((arr - mi) / (ma - mi))
+
+
+# Useful function to generate a Gaussian Function on given coordinates. Used to generate groudtruth.
+def label2maskmap_gt(data, shape, c_dx=0, c_dy=0, radius=10, normalize=False):
+    """
+    Generate a Mask map from the coordenates
+    :param shape: dimension of output
+    :param data : input image
+    :param radius: is the radius of the gaussian function
+    :param normalize : bool for normalization.
+    :return: a MxN normalized array
+    """
+
+    # Our 2-dimensional distribution will be over variables X and Y
+    (M, N) = (shape[2], shape[1])
+    if len(data) <= 2:
+        # Output coordinates are reduced during post processing which poses a problem
+        data = [0, data[0], data[1]]
+    maskMap = []
+
+    x, y = data[2], data[1]
+
+    # Correct the labels
+    x += c_dx
+    y += c_dy
+
+    X = np.linspace(0, M - 1, M)
+    Y = np.linspace(0, N - 1, N)
+    X, Y = np.meshgrid(X, Y)
+
+    # Pack X and Y into a single 3-dimensional array
+    pos = np.empty(X.shape + (2,))
+    pos[:, :, 0] = X
+    pos[:, :, 1] = Y
+
+    # Mean vector and covariance matrix
+    mu = np.array([x, y])
+    Sigma = np.array([[radius, 0], [0, radius]])
+
+    # The distribution on the variables X, Y packed into pos.
+    Z = multivariate_gaussian(pos, mu, Sigma)
+
+    # Normalization
+    if normalize:
+        Z *= (1 / np.max(Z))
+    else:
+        # 8bit image values (the loss go to inf+)
+        Z *= (1 / np.max(Z))
+        Z = np.asarray(Z * 255, dtype=np.uint8)
+
+    maskMap.append(Z)
+
+    if len(maskMap) == 1:
+        maskMap = maskMap[0]
+
+    return np.asarray(maskMap)
+
+
+def multivariate_gaussian(pos, mu, Sigma):
```

Comment:
same as comment above, that could probably be imported from existing Python packages

Reply:
Done in 839deb4